contracts/protocol/libraries/logic/LiquidationLogic.sol

```diff
@@ -329,6 +228,203 @@ library LiquidationLogic {
     );
   }
 
+  /**
+   * @notice Burns the collateral aTokens and transfers the underlying to the liquidator.
+   * @param collateralReserve The data of the collateral reserve
+   * @param params The additional parameters needed to execute the liquidation function
+   * @param vars the executeLiquidationCall() function local vars
+   */
+
+  function _burnCollateralATokens(
+    DataTypes.ReserveData storage collateralReserve,
+    DataTypes.ExecuteLiquidationCallParams memory params,
+    LiquidationCallLocalVars memory vars
+  ) internal {
+    DataTypes.ReserveCache memory collateralReserveCache = collateralReserve.cache();
+    collateralReserve.updateState(collateralReserveCache);
+    collateralReserve.updateInterestRates(
+      collateralReserveCache,
+      params.collateralAsset,
+      0,
+      vars.maxCollateralToLiquidate
+    );
+
+    // Burn the equivalent amount of aToken, sending the underlying to the liquidator
+    vars.collateralAToken.burn(
+      params.user,
+      msg.sender,
+      vars.maxCollateralToLiquidate,
+      collateralReserveCache.nextLiquidityIndex
+    );
+  }
+
+  /**
+   * @notice Liquidates the user aTokens by transferring them to the liquidator.
+   * @param reservesData The state of all the reserves
+   * @param reservesList The addresses of all the active reserves
+   * @param usersConfig The users configuration mapping that track the supplied/borrowed assets
+   * @param collateralReserve The data of the collateral reserve
+   * @param params The additional parameters needed to execute the liquidation function
+   * @param vars the executeLiquidationCall() function local vars
+   */
+
+  function _liquidateATokens(
```

Comment:
Practically, it might not make a difference because of low interest rates. But the state and interest rates should probably be updated even when using `_liquidateATokens()` as the utilization changes because of the interest accrued. Fine to keep as is as it will behave similar to transfer which is essentially is.

Reply:
yes, for consistency with the classic transfer since it's just a transfer of aTokens (nothing change in the underlying) i would keep it this way